afqinsight/plot.py

```diff
@@ -331,3 +333,85 @@ def plot_tract_profiles(
         figs[metric] = fig
 
     return figs
+
+
+def plot_profiles_by_group(
+    tracto_df:pd.DataFrame,
+    var_df:pd.DataFrame,
+    center_cut=True,
+    test_plot=False,
+    scalar_to_plot='dki_fa',
+    var_to_plot:str=None,
+    palette='viridis',
+    ):
+    """Plot tractometry profiles grouped by variable of interest
+
+    Parameters
+    ----------
+    tracto_df : pd.DataFrame
+        tractometry data
+    var_df : pd.DataFrame
+        dataframe with categorical or pseudocontinuous variable of interest
+    center_cut : bool, optional
+        whether to cut the extremes of the tract out, by default True
+    test_plot : bool, optional
+        whether to subsample the data to get quick plot, by default False
+    scalar_to_plot : str, optional
+        scalar to plot (y axis), by default 'dki_fa'
+    var_to_plot : str, optional
+        variable of interest name if `var_df` contains several variables, by default None
+
+    Returns
+    -------
+    sns.FacetGrid
+        FacetGrid
+    """
+    
+    if var_to_plot is None:
+        available = list(set(var_df.columns) - set(['subjectID','sessionID']))
+        if len(available)>1:
+            raise ValueError(f"Please specify `var_to_plot` as there are "
+                             f"several options available in `var_df`.")
+        var_to_plot = available[0]
+    else:
+        if var_to_plot not in var_df.columns:
+            raise ValueError(f"{var_to_plot=} not available in `var_df` columns.")
```

Comment:
I think the variable expansion in the f-string is a syntax error, which is what fails in the CI

```suggestion
            raise ValueError(f"{var_to_plot} not available in `var_df` columns.")
```

Reply:
Please edit [this file](https://github.com/yeatmanlab/AFQ-Insight/blob/main/.github/workflows/test.yml#L11) and [this file](https://github.com/yeatmanlab/AFQ-Insight/blob/main/.github/workflows/docbuild.yml#L11) to remove python version 3.7, so that this syntax can be used.